src/type_.rs

```diff
@@ -125,6 +125,12 @@ impl<'gcc, 'tcx> BaseTypeMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
         else if typ.dyncast_vector().is_some() {
             TypeKind::Vector
         }
+        else if typ.is_compatible_with(self.float_type) {
+            TypeKind::Float
+        }
+        else if typ.is_compatible_with(self.double_type) {
+            TypeKind::Double
+        }
```

Comment:
Please rebase or remove this if not needed anymore. Your code might actually be better than what I wrote because I just did a comparison instead of calling `is_compatible_with()` so feel free to keep yours!

Reply:
Rebased and removed the comparison in favor of `is_compatible_with()`. The comparison does a pointer comparison. I don't know if libgccjit guarantees that there is only a single float and double type value.

Reply:
Well, yes, but a float aligned won't have the same pointer as normal float, so it's best to use `is_compatible_with()`.